File: 2_Periodo/AED_II/TPS/Palindromo.java
/**
 * Palindromo
 * @author Vinicius Francisco da Silva
 * @version 1 08/2017
 */
public class Palindromo{
	public static void main(String[] args){
		String str;
		for(str = MyIO.readLine(); !igual(str); str = MyIO.readLine()){
			if(ehPalindromo(str)){
				MyIO.println("SIM");
			}// End if
			else{
				MyIO.println("NAO");
			}// End else
		}// End for
	}// End main()

	/*
     * Metodo publico iterativo para comparar a String
     * e retornar se ela é um palindromo ou não
     * @param String com a palavra que seja comparada
     * para descobrir se é ou não um palindromo
     * @return boolean identificando se é um palindromo
     * ou não
     */
	public static boolean ehPalindromo(String str){
		int i = 0;
		int j = str.length()-1;
		boolean resp = false;
		while(i < j && j > i){
			if(str.charAt(i) == str.charAt(j)){
				resp = true;
			}// End if
			else{
				return false;
			}// End else
			i++;
			j--;
		}// End while
		return resp;
	}// End ehPalindromo()
	
	/*
     * Metodo publico iterativo para comparar a String
     * recebida como parametro com a String "FIM"
     * @param String com a palavra que seja comparada
     * @return boolean caso o parametro seja igual a
     * String "FIM"
     */
	public static boolean igual(String str){
		boolean resp = false;
		if(str.length() != 3){
			return resp;
		}// End if
		if(str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'){
			resp = true;
		}// End if
		return resp;	
	}// End igual()
}// End class Palindromo

File: 2_Periodo/AED_II/TP02/Palindromo em C/Palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
bool igual(char* str);
bool Palindromo(char* str);
bool isPalindromo(char* str,int i,int j,bool resp);
int main(){
	char str[10000];
	for(scanf("%[^\n]\n",str);!igual(str);scanf("%[^\n]\n",str)){
		bool resp = Palindromo(str);
		if(resp == true){
			printf("SIM\n");
		}// End if
		else{
			printf("NAO\n");
		}// End else
		//setbuf(stdin, NULL);
	}// End for
	return 0;
}// End main()

bool Palindromo(char* str){
	return isPalindromo(str,0,strlen(str)-1,false);
}// End bool

bool isPalindromo(char* str,int i,int j,bool resp){
	if(i < j && j > i){
		if(str[i] == str[j]){
			resp = true;
			resp = isPalindromo(str,i+1,j-1,resp);
		}// End if
		else{
			return false;
		}// End else
	}// End if
	return resp;
}// End ehPalindromo()

bool igual(char str[]){
	bool resp = false;
	if(strlen(str) == 3){
		if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M'){
			resp = true;
		}// End if
		else{
			return false;
		}// End else
	}// End if
		else{
			return false;
		}// End else	
		return resp;
}// End igual()
